force-app/main/default/lwc/participantSelector/participantSelector.js

```diff
@@ -371,6 +371,11 @@ export default class ParticipantSelector extends LightningElement {
 
     handleDeselectParticipant(event) {
         if (event) {
+            //filter availableEngagementRows to remove program engagement that we deselected so there are no duplicates
+            this.availableEngagementRows = this.availableEngagementRows.filter(
```

Comment:
If I'm reading this right, this is basically just preventing duplicates - here on line 374-378 you are removing any availableEngagementRows that match the deselected participant, and then down in 389-392 you are adding the deselected participant to the availableEngagementRows list from the event.detail.row fired from the deselected participant?

Reply:
Yeah that is correct @ptewson-sfdo